### PR TITLE
feat: Add CockroachDB module

### PR DESCRIPTION
**Pull request by rafaumlemos**
## What does this PR do?

Introduces Testcontainers.CockroachDB to the project

## Why is it important?

It's important to support CockroachDB

## Related issues
-

## How to test this PR

I implemented the same test logic used by PostgreSQL as a reference

## Follow-ups

The implementation closely resembles that of PostgreSQL

### CODE REVIEW

**Comment by rafaumlemos** [comment]
Ready for review.

Thanks @HofmeisterAn for some extra help.

File: src/Testcontainers.CockroachDb/CockroachDbBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    {
+        return base.Init()
+            .WithImage(CockroachDbImage)
+            .WithPortBinding(CockroachDbPort, true)
+            .WithPortBinding(CockroachDbRestPort, true)
+            .WithDatabase(DefaultDatabase)
+            .WithUsername(DefaultUsername)
+            .WithPassword(DefaultPassword)
+            .WithCommand("start-single-node")
+            .WithCommand("--insecure")
```
Comment (eddumelendez) [inline src/Testcontainers.CockroachDb/CockroachDbBuilder.cs:92]: `--insecure` is not needed if credentials are set

```suggestion
```

Comment (rafaumlemos) [reply src/Testcontainers.CockroachDb/CockroachDbBuilder.cs:92]: I'm thinking to adopt a similar approach as demonstrated in the [CRDB Java](https://github.com/testcontainers/testcontainers-java/blob/main/modules/cockroachdb/src/main/java/org/testcontainers/containers/CockroachContainer.java#L50) repository and use an empty password string while keeping the `---insecure` connection.
What is your opinion on this?

**Comment by HofmeisterAn** [comment]
FYI: I have not forgotten to review the PR. I will do it at the latest when I am back from the holidays.

File: src/Testcontainers.CockroachDb/CockroachDbBuilder.cs (hunk 1 above)
Comment (eddumelendez) [reply src/Testcontainers.CockroachDb/CockroachDbBuilder.cs:92]: sorry for the late reply. Sure, that make sense!

Comment (rafaumlemos) [reply src/Testcontainers.CockroachDb/CockroachDbBuilder.cs:92]: Done!
@eddumelendez

**Comment by rafaumlemos** [comment]
@HofmeisterAn @eddumelendez 

Ready for review again; I've resolved the conflicts due to the new changes in the develop branch.

**Comment by HofmeisterAn** [comment]
@rafaumlemos, sorry for the delay. I am very busy right now. I just had the time to review your pull request. The PR looks good. Thanks for the contribution 🥳! I will merge the PR later.

**Review by HofmeisterAn — APPROVED** [review]

**Pull request merged**